hls4ml/templates/oneapi/firmware/nnet_utils/nnet_depthconv2d_resource.h

```diff
@@ -0,0 +1,61 @@
+#ifndef NNET_SEPARABLE_CONV2D_LATENCY_H_
+#define NNET_SEPARABLE_CONV2D_LATENCY_H_
+
+#include "nnet_common.h"
+#include "nnet_conv2d_resource.h"
+#include "nnet_mult.h"
+
+namespace nnet {
+
+template <class data_T, class res_T, typename CONFIG_T>
+void depthwise_conv_2d_resource_cl(const data_T &data, res_T &res, const typename CONFIG_T::weight_t &weights,
+                                   const typename CONFIG_T::bias_t &biases) {
+
+    int depth_multiplier = CONFIG_T::n_filt / CONFIG_T::n_chan;
+    [[intel::fpga_register]] int res_idx = 0;
+
+DM_LOOP:
+    #pragma unroll
+    for (int dm = 0; dm < depth_multiplier; dm++) {
+
+    CHAN_LOOP:
```

Comment:
If it's channels last, isn't it better to do the channels loop later? Maybe it doesn't matter for HLS, though. Same with the depth multiplier, it might result in fewer data reads if the depth multiplier loop is later. Whether that is actually true I am not sure.